00-introduction.adoc

```diff
@@ -1,199 +1,265 @@
-
-If something awfully complex pops up like geocoding, most folks will
-just shove Google Maps API in there, or Mapbox API, or OpenStreet Map
-API, because generating a standard interface for wildly different
-datasets around the globe is a mess, and why waste time on that when
-you’ve got underpants to fold!
-
-== Breaking Down the Monolith
-
-Over the last few years, the rise of the service (or micro service) has
-meant you’ll often be working with more than one API. External APIs
-might be replaced with internal APIs as your teams grow and ownership
-becomes more important, and larger applications can start to become an
-unmanageable non-performant mess. This really makes sense for companies
-with larger engineering teams, who want to avoid stepping on each others
-toes with changes they make. The more services at play, the more complex
-everything gets, but a mature, and educated engineering department with
-a strong devops culture can eventually learn to manage this.
-
-A lot of teams who are just getting started with building services kinda
-forget about this integration part, and forget their systems are going
-to be used by a myriad of others. As companies grow, other developers
-want to start hooking into these data sources, and often the original
-developers have moved on. You end up with all these random things
-talking to other random things and the architectural diagrams (if kept
-up to date) would look like an octopus orgy.
-
-These services need to talk to each other intelligently, handle various
-types of error, have realistic timeouts, know when to retry, and most
-importantly identify themselves, so you don’t end up with a stampeding
-herd; with no idea which client is causing it.
-
-These services also need frontends, or have one mega dashboard that
-handles the UI for many of them.
-
-One of these scenarios is where you probably come in, so with the why
-covered, let’s look at how.
+production issues. If it's not as dire as that, there are at least other
+sources of confusion that'll get you spinning your wheels trying to
+figure out what to do next. 
+
+Throughout the 2000s, most of us web developers were building frontend and the
+backend in the same application. The frontend was just whatever HTML the backend
+decided to spit out. Frontend developers would design the templates (HTML + data
+tags), and the backend would decide which HTML template to show, and what values
+to shove into those tags. Go Smarty.
+
+It was a simple time, but there was a lot of difficulty and hair-pulling.
+Sharing data between the presentation layer and the the backend was bad enough,
+but sharing data between different organizations was a mess. There were various
+complicated things floating around like SOAP ("Simple Object Access Protocol"),
+but instead of diving into that there was a scary amount of nonsense like iframe
+trickery. 
+
+There was a whole lot of scraping websites and yoinking out data with regex. One
```

Comment:
maybe: add a parenthetical explaining what `regex` is to someone who hasn't encountered it before